src/ocean/text/Util.d

```diff
@@ -785,6 +788,136 @@ bool isSpace(T) (T c)
            return (c <= 32 && (c is ' ' || c is '\t' || c is '\r' || c is '\n' || c is '\f' || c is '\v')) || (c is '\u2028' || c is '\u2029');
 }
 
+/*******************************************************************************
+
+    Enumeration of the different ways in which a string can be padded in order
+    to increase its length.
+
+*******************************************************************************/
+
+public enum PadType : ubyte
+{
+    PadPre,    // Padding is only applied before the string
+    PadPost,   // Padding is only applied after the string
+    PadAround  // Padding is applied on both sides of the string
+}
+
+/*******************************************************************************
+
+    Shortens/Lengthens a string as necessary to get it to a desired length.
+
+    Note 1:
+        If the string is shortened, an ellipsis (3 dots) will be appended to
+        indicate intentional truncation.
+
+    Note 2:
+        The length specified must be at least 4. If this constraint were absent
+        and a length of 3 or lower was specified, then the input string would be
+        converted simply to an ellipsis, losing all information about the
+        string.
+
+    Note 3:
+        If the input string needs to be lengthened and padding on both sides
+        is requested; then the padding will be added evenly on both sides if the
+        input string length and the desired length are both even or both odd,
+        otherwise an extra space will be added on the right side.
+
+    Params:
+        orig_str = original input string
+        final_length = desired length of the final string
+        final_str = transformed output string
+        pad_type = defines how the string should be padded (applicable only if
+            the string needs to be lengthened, defaults to padding being added
+            after the original string)
+
+    Returns:
+        The transformed string with the desired length
+
+*******************************************************************************/
+
+public mstring setLength (cstring orig_str, size_t final_length,
```

Comment:
>It's more typical for such functions in ocean to accept a ref mstring and mutate it in-place. What was the reason for rendering into a separate buffer?

The main reason is that it makes the calling code simpler, as the caller doesn't need to copy the string into a mutable buffer first, and only needs to provide a mutable buffer. The other thing is that if we need to lengthen the string by adding the padding characters _before_ it, then mutating it in place would be a bit unwieldy and you'd probably need to use a temporary static buffer to make this happen.

>I find the name setLength a little deceptive. It sounds like a simple `array.length = X;` wrapper.

Yes, I agree. How about `strSetLength()` or `setStrLength()`? I'm open to other suggestions as well.